core/jobs/transforms/filter_results_transforms.py

```diff
@@ -0,0 +1,97 @@
+# limitations under the License.
+
+"""Provides an Apache Beam API for operating on NDB models."""
+
+from __future__ import annotations
+
+import apache_beam as beam
+import result
+from typing import Any, Optional, Tuple
+
+
+# TODO(#15613): Here we use MyPy ignore because the incomplete typing of
+# apache_beam library and absences of stubs in Typeshed, forces MyPy to
+# assume that PTransform class is of type Any. Thus to avoid MyPy's error
+# (Class cannot subclass 'PTransform' (has type 'Any')), we added an
+# ignore here.
+class FilterResults(beam.PTransform):  # type: ignore[misc]
+    """Transform that flushes an input PCollection if any error results
+       are encountered..
+    """
+
+    def __init__(
+        self, prefix: Optional[str] = None, label: Optional[str] = None
+    ) -> None:
+        """Initializes the FilterResults PTransform.
+
+        Args:
+            prefix: str|None. The prefix for the result string.
+            label: str|None. The label of the PTransform.
+        """
+        super().__init__(label=label)
+        self.prefix = '%s ' % prefix if prefix else ''
+
+    # Here we use type Any because this method can accept Result object
+    # with any type of domain object.
+    @staticmethod
+    def _check_migration_errors(
+        unused_result: result.Result[Tuple[str, Any], Tuple[str, Exception]],
+        is_no_migration_error: beam.pvalue.AsSingleton
+    ) -> bool:
+        """Checks if any errors have occured.
+
+        Args:
+            unused_result: result.Result. Unused Result object.
+            is_no_migration_error: beam.pvalue.AsSingleton. Side input data
+                specifying non-zero erros during migration.
+
+        Returns:
+            bool. Specifies whether any migration errors were found.
+        """
+        return bool(is_no_migration_error)
+
+    # Here we use type Any because this method can accept any kind of
+    # PCollection object to return the filtered migration results.
+    def expand(
+        self, objects: beam.PCollection[result.Result[Tuple[str, Any], Tuple[str, Exception]]] # pylint: disable=line-too-long
+    ) -> beam.PCollection[result.Result[Tuple[str, Any], None]]:
+        """Count error results in collection and flush the input
+            in case of errors.
+
+        Args:
+            objects: PCollection. Sequence of Result objects.
+
+        Returns:
+            PCollection. Sequence of Result objects or empty PCollection.
+        """
+
+        error_check = (
+            objects
+            | 'Filter errors' >> beam.Filter(
+                lambda result_item: result_item.is_err())
+            | 'Count number of errors' >> beam.combiners.Count.Globally()
+            | 'Check if error count is zero' >> beam.Map(lambda x: x == 0)
+        )
+
+        filtered_results = (
+            objects
+            | 'Remove all results in case of errors' >> beam.Filter(
+                self._check_migration_errors,
```

Comment:
Can this be simplified by doing:
```suggestion
                lambda _, no_migration_error: bool(no_migration_error)
```

Reply:
Done